src/webview_provider/altimateWebviewProvider.ts

```diff
@@ -188,6 +188,15 @@ export class AltimateWebviewProvider implements WebviewViewProvider {
 
     try {
       switch (command) {
+        case "getProjectAdapterType":
+          this.handleSyncRequestFromWebview(
+            syncRequestId,
+            () => {
+              return this.queryManifestService.getProject()?.getAdapterType();
```

Comment:
Can you guarantee this is the same project as on load. I assume the getProject will check the selected text editor again, what if we want to implement a onSelectedTextEditor handler that deals with unsaved data. You would have to fix all these occurrences.

Reply:
This command is executed when user selects query translate and land in datapilot panel. So user will have the file open in editor and we find the adapter type based on this data. So, I think this should be fine

Reply:
Why not assemble all this information at the start? I think this is more complicated than necessary.

Reply:
This will need changes in multiple places, i will address this in separate PR. Created a ticket here: https://altimateai.atlassian.net/browse/AI-1571